python/lsst/meas/astrom/astrometry.py

```diff
@@ -358,16 +377,64 @@ def _matchAndFitWcs(self, refCat, sourceCat, goodSourceCat, refFluxField, bbox,
             displayAstrometry(
                 refCat=refCat,
                 sourceCat=matchRes.usableSourceCat,
-                matches=matchRes.matches,
+                matches=matches,
                 exposure=exposure,
                 bbox=bbox,
                 frame=frame + 2,
                 title="Fit TAN-SIP WCS",
             )
 
         return pipeBase.Struct(
-            matches=matchRes.matches,
+            matches=matches,
             wcs=fitWcs,
             scatterOnSky=scatterOnSky,
             match_tolerance=matchRes.match_tolerance,
         )
+
+    def _removeMagnitudeOutliers(self, sourceFluxField, refFluxField, matchesIn):
+        """Remove magnitude outliers, computing a simple zeropoint.
+
+        Parameters
+        ----------
+        sourceFluxField : `str`
+            Field in source catalog for fluxes.
+        refFluxField : `str`
+            Field in reference catalog for fluxes.
+        matchesIn : `list` [`lsst.afw.table.ReferenceMatch`]
+            List of source/reference matches input
+
+        Returns
+        -------
+        matchesOut : `list` [`lsst.afw.table.ReferenceMatch`]
+            List of source/reference matches with magnitude
+            outliers removed.
+        """
+        nMatch = len(matchesIn)
+        sourceMag = np.zeros(nMatch)
+        refMag = np.zeros(nMatch)
+        for i, match in enumerate(matchesIn):
+            sourceMag[i] = -2.5*np.log10(match[1][sourceFluxField])
+            refMag[i] = (match[0][refFluxField]*units.nJy).to_value(units.ABmag)
+
+        deltaMag = refMag - sourceMag
+        # Protect against negative fluxes and nans in the reference catalog.
+        goodDelta, = np.where(np.isfinite(deltaMag))
+        zp = np.median(deltaMag[goodDelta])
+        # Use median absolute deviation (MAD) for zpSigma
+        # Also require a minimum scatter to prevent floating-point errors from
+        # rejecting objects in zero-noise tests.
+        zpSigma = np.clip(1.4826*np.median(np.abs(deltaMag[goodDelta] - zp)), 1e-3, None)
+
+        self.log.debug(f"Rough zeropoint from astrometry matches is {zp:.4f} +/- {zpSigma:.4f}.")
```

Comment:
Just a fly-by comment...I'd actually like to see this as a `log.info` since it could be very useful to know in case things went horribly "wrong" with this zeropoint estimate (i.e. most likely something amiss with the source catalog itself).  We should normally have a "decent" estimate of what a photometric zeropoint should be, so having this logged here might help reveal seriously non-photometric visits (or something else funky going on).

Reply:
I don't think that should be something we have to extract from the logs: that's a number that we should be recording as a metric (much like the astrometric scatter from the fit), so that we can readily access and plot it.

Reply:
Please use the lazy-logging formatter for, so that the string doesn't have to be evaluated unless it is logged, e.g. `log.debug("Rough zeropoint from astrometry matches is %.4f +/- %.4f.", zp, zpSigma)`